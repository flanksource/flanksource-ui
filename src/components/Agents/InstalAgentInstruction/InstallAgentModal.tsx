import { useState } from "react";
import { GeneratedAgent } from "../../../api/services/agents";
import { Button } from "../../../ui/Buttons/Button";
import { Modal } from "../../../ui/Modal";
import { Tab, Tabs } from "../../../ui/Tabs/Tabs";
import { AgentFormValues } from "../Add/AddAgentForm";
import CLIInstallAgent from "./CLIInstallAgent";
import FluxInstallAgent from "./FluxInstallAgent";

export function WarningBox() {
  return (
    <div
      className="bg-yellow-100 border border-yellow-200 text-yellow-700 px-4 py-3 rounded relative"
      role="alert"
    >
      <span className="block sm:inline">
        Access token will be shown only once. Please copy it and store it
        securely.
      </span>
    </div>
  );
}

export function MoreInfoBox() {
  return (
    <div className="flex flex-col gap-2 px-2">
      <p>
        See the{" "}
        <a
          className="text-blue-500 hover:underline"
          href="https://docs.flanksource.com/registry/kubernetes/"
          target="_blank"
          rel="noreferrer"
        >
          Kubernetes
        </a>{" "}
        for more options on fine-tuning the import of Kubernetes resources.
      </p>

      <h3 className="font-bold text-lg">Next Steps</h3>

      <ul className="list-disc px-6">
        <li>
          Review the dashboard and catalog pages to verify the data being
          imported.
        </li>
        <li>
          Create new{" "}
          <a
            href="https://docs.flanksource.com/canary-checker/getting-started/"
            target="_blank"
            rel="noreferrer"
            className="text-blue-500 hover:underline"
          >
            health checks
          </a>{" "}
          and{" "}
          <a
            target="_blank"
            className="text-blue-500 hover:underline"
            href="https://docs.flanksource.com/config-db/overview"
            rel="noreferrer"
          >
            catalog scrapers
          </a>{" "}
          using Gitops in the agent cluster.{" "}
        </li>
        <li>
          Create new{" "}
          <a
            href="https://docs.flanksource.com/topology/overview"
            target="_blank"
            rel="noreferrer"
            className="text-blue-500 hover:underline"
          >
            topologies
          </a>{" "}
          to model your system -Topologies can be created either in the agent
          cluster or centrally to create views that span multiple agents.
        </li>
      </ul>
    </div>
  );
}

type Props = {
  isOpen: boolean;
  onClose: () => void;
  generatedAgent: GeneratedAgent;
  agentFormValues?: AgentFormValues;
};

export default function InstallAgentModal({
  isOpen,
  onClose,
  generatedAgent,
  agentFormValues
}: Props) {
  const [activeTab, setActiveTab] = useState<"cli" | "flux">("cli");

  return (
    <Modal
      title={"Installation Instructions"}
      onClose={onClose}
      open={isOpen}
      bodyClass="flex flex-col w-full flex-1 h-full overflow-y-auto"
    >
      <div className="flex flex-col gap-4 flex-1 p-4 overflow-y-auto">
        <h3 className="font-bold text-lg">
          Install the Mission Control agent using instructions below
        </h3>
        <div className="flex flex-col ">
          <Tabs
            activeTab={activeTab}
            onSelectTab={(v) => setActiveTab(v as any)}
          >
            <Tab className="flex flex-col gap-4 p-4" label="Flux" value="flux">
              <FluxInstallAgent
                generatedAgent={generatedAgent}
                agentFormValues={agentFormValues}
              />
              <WarningBox />
            </Tab>
            <Tab
              className="flex flex-col gap-4 p-4"
              label="Helm CLI"
              value="cli"
            >
              <CLIInstallAgent
                generatedAgent={generatedAgent}
                agentFormValues={agentFormValues}
              />
              <WarningBox />
            </Tab>
          </Tabs>
        </div>
        <MoreInfoBox />
      </div>
      <div className="flex flex-row justify-end gap-4 p-4">
        <Button text="Close" onClick={onClose} />
      </div>
    </Modal>
  );
}
