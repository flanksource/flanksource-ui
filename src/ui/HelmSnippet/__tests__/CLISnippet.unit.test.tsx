import { render, screen } from "@testing-library/react";
import InstallAgentModal from "../CLISnippet";
import { mockInput, mockInputWithKubOptions } from "./mocks/mocks";

global.ResizeObserver = jest.fn().mockImplementation(() => ({
  observe: jest.fn(),
  unobserve: jest.fn(),
  disconnect: jest.fn()
}));

describe("InstallAgentModal", () => {
  it("renders the Helm repository installation command", () => {
    render(<InstallAgentModal data={[mockInput]} />);
    const element = screen.getByText((content, element) => {
      return (
        element?.tagName.toLowerCase() === "pre" &&
        element?.textContent?.includes("helm repo add flanksource") === true
      );
    });
    expect(element.textContent).toMatchSnapshot();
  });

  it("renders the Helm repository installation command with kube command", () => {
    render(<InstallAgentModal data={mockInputWithKubOptions} />);
    const element = screen.getByText((content, element) => {
      return (
        element?.tagName.toLowerCase() === "pre" &&
        element?.textContent?.includes("helm repo add flanksource") === true
      );
    });
    expect(element.textContent).toMatchSnapshot();
  });
});
