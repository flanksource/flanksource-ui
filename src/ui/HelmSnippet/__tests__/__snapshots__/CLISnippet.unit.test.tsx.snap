// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstallAgentModal renders the Helm repository installation command 1`] = `
"helm repo add flanksource https://flanksource.github.io/charts
helm install mission-control-agent flanksource/mission-control-agent -n "mission-control-agent" \\  --set upstream.createSecret=true  \\   --set upstream.host=http://localhost:3000  \\   --set upstream.username=token  \\   --set upstream.password=password  \\   --set upstream.agentName=test-new-agent-instructions  \\   --set pushTelemetry.enabled=true  \\   --set pushTelemetry.topologyName=incident-commander.demo.aws.flanksource.com-test-new-agent-instructions  \\  --create-namespace    
"
`;

exports[`InstallAgentModal renders the Helm repository installation command with kube command 1`] = `
"helm repo add flanksource https://flanksource.github.io/charts
helm install mission-control-agent flanksource/mission-control-agent -n "mission-control-agent" \\  --set upstream.createSecret=true  \\   --set upstream.host=http://localhost:3000  \\   --set upstream.username=token  \\   --set upstream.password=password  \\   --set upstream.agentName=test-new-agent-instructions  \\   --set pushTelemetry.enabled=true  \\   --set pushTelemetry.topologyName=incident-commander.demo.aws.flanksource.com-test-new-agent-instructions  \\  --create-namespace    
helm install mission-control-kubernetes flanksource/mission-control-kubernetes -n "mission-control-agent" \\  --set clusterName=test-new-agent-instructions2  \\   --set scraper.schedule=@every 31m    
"
`;
